### ISSUE TITLE
auto finish task using AD

### ISSUE THREAD

**Issue by RottenMind** [Phr33d0m/NW-Profession-Bot#49]
, some other gateway bots (if I remember right ) has done samekind things. I think its similar with "start 4th task" issue, Gateway-client lag or something.

(edit, abandoned APB/APBI did same thing, when we tested them, I think Cyrealkiller added some waittimes, not sure, but it was rare issue, year ago)

https://github.com/Phr33d0m/NW-Profession-Bot/issues/17
https://github.com/Phr33d0m/NW-Profession-Bot/issues/46

from Greasyfork,

https://greasyfork.org/en/forum/discussion/2973/x

<blockquote class="Quote" rel="NineZero">First thanks for your hard work!
Im not leveling leadership actualy, and i found that im losing small amounts of AD realy small. i think somethimes the script dont wait until the very end(maybe lag) and it auto finish it using AD. Since is small time left it spends realy small amounts.

Is there a way to increase the time the bot should take to collect it in order to prevent this?
Thanks you.</blockquote>

**Comment by Phr33d0m** [comment]
Is this a a bug report or a feature request?

If it's a bug report, from what I saw the bot has literally no code to force-complete a task with AD.

**Comment by RottenMind** [comment]
I must clarify, read text again, maybe you forgot word "rare" and "lag", maybe it is feature request that Bot checks better "Start Task"  button condition and wait longer so if lag exist and  it will not click button prematurely.

...and go ask user who reported it. Users not make Github account for bug reports, they stay Greasyfork and Ownedcore, :D.

**Comment by noonereally** [comment]
As far as i understand the code, it doesn't work like that, the code doesn't set a timer and then press a button.
The script sets a timer to switch to character.
Switch when it can.
Looks for 'finished task' buttons !!
Press them. 

Even if you have a few days of lag, it is not important, the script doesn't time the button press, it only press **already finished** tasks.

**Comment by RottenMind** [comment]
@noonereally , bots I referring use geckofx and thier sourcecode is lost. 

Today I saw when Bot tried buy ingredients, it did Buy and log says succeed, but no items inventory. Stopped script and checked gateway,  found nothing special, started script and it continues buying nothing. Other bots I have acted as they do normally. After page reload problem resolved.

_So it is gateway bug and bot fails to recover or detect it_... (nothing can do, but set page reload per 2 hours with external script and AD task finishing is really rare)
